### PR TITLE
usb: sam0: Add support for SAME54

### PR DESCRIPTION
**Pull request by benpicco**
The USB peripheral is the same as in previous SAM0 MCUs, but the clock setup is different.

split off from #14685

### CODE REVIEW

File: drivers/usb/device/usb_dc_sam0.c [hunk 1]
```diff
@@ -132,6 +132,9 @@ static void usb_sam0_load_padcal(void)
 	u32_t pad_transp;
 	u32_t pad_trim;
 
+#ifdef USB_FUSES_TRANSN_ADDR
+	pad_transn = *(uint32_t *)USB_FUSES_TRANSN_ADDR;
```
Comment (jfischer-phytec-iot) [inline drivers/usb/device/usb_dc_sam0.c]: This change is not explained in the commit message. Where is USB_FUSES_TRANSN_ADDR defined, HAL?

File: drivers/usb/device/usb_dc_sam0.c [hunk 2]
```diff
@@ -203,10 +235,18 @@ int usb_dc_attach(void)
 	regs->INTENSET.reg = USB_DEVICE_INTENSET_EORST;
 
 	/* Connect and enable the interrupt */
-	IRQ_CONNECT(DT_INST_0_ATMEL_SAM0_USB_IRQ_0,
-		    DT_INST_0_ATMEL_SAM0_USB_IRQ_0_PRIORITY,
-		    usb_sam0_isr, 0, 0);
-	irq_enable(DT_INST_0_ATMEL_SAM0_USB_IRQ_0);
+#ifdef DT_INST_0_ATMEL_SAM0_USB_IRQ_0
```
Comment (jfischer-phytec-iot) [inline drivers/usb/device/usb_dc_sam0.c]: This change is also not explained in the commit message.

**Review by jfischer-no — CHANGES_REQUESTED** [review]
Please split it in three commits: changes for PAD? settings, changes for clock setup, and changes for IRQ connect and enable.

**Comment by benpicco** [comment]
Ok, I've split up the commit.

File: drivers/usb/device/usb_dc_sam0.c (hunk 1 above)
Comment (benpicco) [reply drivers/usb/device/usb_dc_sam0.c]: Yes, it's in the [`component/nvmctrl.h`](https://github.com/zephyrproject-rtos/hal_atmel/blob/master/asf/sam0/include/same54/component/nvmctrl.h#L743) header file.

**Review by Sizurka — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**